app/controllers/admin/lead_providers_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class Admin::LeadProvidersController < ApplicationController
+  def index
+    @lead_providers = LeadProvider.all
+  end
+
+  def new
+    @lead_provider = LeadProvider.new
+  end
+
+  def create
+    @lead_provider = LeadProvider.new(name: params.dig(:lead_provider, :name))
+
+    if @lead_provider.save
+      redirect_to admin_lead_providers_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @lead_provider = LeadProvider.find(params.require(:id))
+  end
+
+  def update
+    @lead_provider = LeadProvider.find(params.fetch(:id))
+
+    if params[:lead_provider][:name].blank?
+      @lead_provider.errors.add(:name, "Enter a name")
+      render :edit and return
+    end
```

Comment:
This should definitely live in `LeadProvider` model.

Then doing `@lead_provider.update(...)` will give you boolean saying whether it worked or not (I think), and we can use similar pattern to create.